[Feature][docs] add docs for introducing embedding models and vector stores

alnzng opened the pull request:
<!--
* Thank you very much for contributing to Flink Agents.
* Please add the relevant components in the PR title. E.g., [api], [runtime], [java], [python], [hotfix], etc.
-->

<!-- Please link the PR to the relevant issue(s). Hotfix doesn't need this. -->
Linked issue: #204

### Purpose of change

<!-- What is the purpose of this change? -->
add docs for introducing embedding models and vector stores

### Tests

<!-- How is this change verified? -->
<img width="1170" height="505" alt="Screenshot 2025-09-26 at 12 00 39 AM" src="https://github.com/user-attachments/assets/ab7afe11-ebcd-42e6-a0e4-9629ba47c653" />

### API

<!-- Does this change touches any public APIs? -->
n/a

### Documentation

<!-- Should this change be covered by the user documentation?-->
n/a

alnzng commented:
@xintongsong I added docs for embedding models and vector stores based on the existing implementations, please help take a look.

xintongsong commented on docs/content/docs/development/vector_stores.md:
```diff
@@ -0,0 +1,250 @@
+
+    @vector_store_setup
+    @staticmethod
+    def chroma_store() -> Tuple[Type[ChromaVectorStoreSetup], Dict[str, Any]]:
```
Thanks for pointing this out. It makes perfect sense to me.

xintongsong commented:
@alnzng, thanks for the really nice work on these documentations. A few suggestions from my side.

### Embedding Models

In general, I'd suggest the following order to introduce things to our users.
1. How to use embedding models in general
2. What built-in providers are available, and their specifics
3. How to provide custom provider support, in the built-in ones do not meet requirements

In that sense, I'd suggest the following changes.
1. Move the introduction of the connection and setup concepts, and why we have this separation, to the `Overview` section.
2. Change the `Embedding Interfaces` section name into something like `Use Embedding Models in Agents`, move `BaseEmbeddingModelConnection` and `BaseEmbeddingModelSetup` out, and add a general example that not only shows how to define the model connection and setup with the decorators, but also how to obtain the embedding model with `RunnerContext.get_resource()` and call the `embed()` method on it. 
3. Add another section `Support Custom Providers` after `Built-in Providers`, and introduce `BaseEmbeddingModelConnection` and `BaseEmbeddingModelSetup`, which are the interfaces need to be implemented for custom supports.

In addition, I'd suggest to add a notice for the `Support Custom Providers` section, reminding users that the APIs are experimental and unstable, subject to in-compatible changes in future releases.

### Vector Stores

My suggestions are similar to those on Embedding Models, to separate the sections from the user's perspective: how to use it, what built-in supports are available, and how to customize in built-in supports don't meet requirements.

alnzng commented:
Great suggestions, @xintongsong ! I updated the docs, please take another look, thanks.

xintongsong commented:
Thanks for addressing my comment, @alnzng. 

Now I think the documentation reads really well. However, it seems some latest changes are not captured by the sample codes. Have you tried running the sample codes on the latest main branch?
- We refactored the signature of decorated functions. They used to return a class, followed by a list of arguments used for initializing the class. Now we expect the function to return a ResourceDescriptor. See #211 for more details.
- `ctx.get_resource()` should also have an argument of the resource type.

@wenjin272, could you also help checking on the sample codes?

alnzng commented:
@xintongsong Oh, I should use the latest API in the doc. I know we have changed to `ResourceDescriptor` recently, but forgot to update the example codes here. (I copied this from one of my very old example). Thanks for pointing this out! I updated the doc accordingly.

@wenjin272 Please help take a second eye on this, thanks!

wenjin272 commented:
Hi, @alnzng and @xintongsong, I build the docs locally, read the document and check the code.  All things look good to me.
